service/src/main/java/com/epam/esm/service/impl/UserServiceImpl.java

```diff
@@ -40,8 +56,49 @@ public UserDto findUserByIdService(Long id) {
         return userServiceMapper.convertUserToDto(user);
     }
 
+    @Transactional
     @Override
-    public Long count(){
+    public UserDto signup(UserDto userDto) {
+        User user = userServiceMapper.convertUserFromDto(userDto);
+
+        List<ValidationError> validationErrors = userValidator.validate(user);
+
+        if (!validationErrors.isEmpty()) {
+            throw new InvalidEntityException(User.class);
+        }
+
+        if (userRepository.findByUsername(user.getUsername()).isPresent()) {
+            throw new EntityAlreadyExistsException();
+        }
+
+        String encodedPassword = encoder.encode(user.getPassword());
+        user.setPassword(encodedPassword);
+
+        User savedUser = userRepository.save(user);
+        Optional<User> userById = userRepository.findById(savedUser.getId());
+        if (userById.isEmpty()) {
+            throw new EntityNotFoundException(String.valueOf(savedUser.getId()));
+        }
+        return userServiceMapper.convertUserToDto(userById.get());
+    }
+
+    @Override
+    public UserDto login(UserDto userDto) {
```

Comment:
Why the entity of the whole user is used for authentication. It seems to me that it would be better if you create a separate AuthenticationDto

Reply:
fixed